projects/graphql-client/src/utils/builders/request/graphql-request-builder.ts

```diff
@@ -24,23 +27,35 @@ export class GraphQlRequestBuilder {
     return this.keyBySelection.get(selection)!;
   }
 
-  private translateMergedRequestToGql(mergedRequest: MergedRequestFragment): string {
-    const contents = Object.keys(mergedRequest)
-      .map(key => {
-        const value = mergedRequest[key];
-        const valueAsString = isEmpty(value.childFragment)
-          ? ''
-          : ` ${this.translateMergedRequestToGql(value.childFragment)}`;
-        const keyAsString = this.convertKeyToFieldExpression(key, value);
-        const argString = this.convertArgumentMapToString(value.argMap);
-
-        return `${keyAsString}${argString}${valueAsString}`;
+  public buildSelectionToGqlRequestStringMap(): Map<GraphQlSelection, string> {
+    return new Map(
+      this.originalSelections.map(selection => {
```

Comment:
Yes. So to avoid complexity (and to use Apollo's batching), I am sending each selection in its own gql request. The HttpBatchLink batches these selections into individual http request according to our config and returns the results seamlessly.

Reply:
It makes sense to send each query as a separate request and use apollo's batching, but by keying off the original request aren't we negating all merging? I would think we could use the previous logic almost identically, but rather than returning a string return an array of strings (since we know all the top level keys of the merged result in the request build, we have the info we need to split these into appropriate queries)

Reply:
Or is this returning identical queries for many different selections, and apollo is collapsing them?

Reply:
The original selection is only used for response parsing. For example, The TraceHandler requests both trace and span query using a selection map. So this given request has two selection. By maintaining original selection to merged request fragment mapping, we ensure that responses for both merged selection are passed on to the gql extractor for further processing.

In the Graphql Request service, we extract values array from this map, and we only intend to pass unique values to apollo. I just realized that I missed the `unique` call there. I will add it.